python/ray/serve/backend_state.py

```diff
@@ -136,29 +248,29 @@ def create_backend(self, backend_tag: BackendTag,
                        backend_config: BackendConfig,
                        replica_config: ReplicaConfig) -> Optional[GoalId]:
         # Ensures this method is idempotent.
-        backend_info = self.backends.get(backend_tag)
+        backend_info = self._backend_metadata.get(backend_tag)
         if backend_info is not None:
             if (backend_info.backend_config == backend_config
                     and backend_info.replica_config == replica_config):
                 return None
 
-        backend_replica = create_backend_replica(replica_config.func_or_class)
+        backend_replica_class = create_backend_replica(
+            replica_config.func_or_class)
 
         # Save creator that starts replicas, the arguments to be passed in,
         # and the configuration for the backends.
         backend_info = BackendInfo(
-            worker_class=backend_replica,
+            worker_class=backend_replica_class,
             backend_config=backend_config,
             replica_config=replica_config)
 
         new_goal_id, existing_goal_id = self._set_backend_goal(
             backend_tag, backend_info)
 
         try:
-            self.scale_backend_replicas(backend_tag,
-                                        backend_config.num_replicas)
+            self.scale_backend_replicas(backend_tag)
```

Comment:
instead of having separate calls to scale_backend_replicas in each API method we can now just to it in the update loop

Reply:
that way there's clean separation that the API methods just change goal state and the update loop makes requisite changes